components/progress-indicator/index.jsx

```diff
@@ -123,11 +123,16 @@ const defaultProps = {
  * Check if the passed steps are valid
  */
 function checkSteps (steps) {
-	if (steps === undefined) return false;
-	for (let i = 0; i < steps.length; ++i) {
-		if (steps[i].label === undefined) return false;
+	const anyLabelUndefined = (step) => step.label === undefined;
+	let valid = true;
+
+	if (steps === undefined) {
+		valid = false;
+	} else if (Array.isArray(steps) && steps.some(anyLabelUndefined)) {
```

Comment:
If you move this expression and the one above to a local variable and invert the boolean test, then the entire `if/else if` block could be removed, the code could be more readable, and line 135 could be simplified. For example:
```
const labelDefined = (step) => step.label !== undefined;
const stepsDefined = steps !== undefined;
const stepLabelsDefined = Array.isArray(steps) && steps.all(labelDefined);

return stepsDefined && stepLabelsDefined
```

Meaning... you're returning `true` only if _both_ `steps` is defined _and all_ step labels are defined.